src/it/scala/io/iohk/ethereum/sync/RegularSyncItSpec.scala

```diff
@@ -76,6 +95,91 @@ class RegularSyncItSpec extends FreeSpecBase with Matchers with BeforeAndAfterAl
     }
   }
 
+  "peers should keep being synced on checkpoints" in customTestCaseResourceM(
+    FakePeer.start2FakePeersRes()
+  ) { case (peer1, peer2) =>
+    val blockNumber: Int = 2000
+    for {
+      _ <- peer1.importBlocksUntil(blockNumber)(IdentityUpdate)
+      _ <- peer1.startRegularSync()
+      _ <- peer2.startRegularSync()
+      _ <- peer2.connectToPeers(Set(peer1.node))
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumber)
+      _ <- peer2.addCheckpointedBlock(peer2.bl.getBestBlock().get)
+      _ <- peer1.waitForRegularSyncLoadLastBlock(blockNumber + 1)
+    } yield {
+      assert(peer1.bl.getBestBlock().get.hash == peer2.bl.getBestBlock().get.hash)
+      assert(peer1.bl.getLatestCheckpointBlockNumber() == peer2.bl.getLatestCheckpointBlockNumber())
+    }
+  }
+
+  "peers should keep being synced on checkpoints even if 2 checkpoints are issued to different forks at the same time" in customTestCaseResourceM(
+    FakePeer.start2FakePeersRes()
+  ) { case (peer1, peer2) =>
+    val blockNumber: Int = 2000
+    for {
+      _ <- peer1.importBlocksUntil(blockNumber)(IdentityUpdate)
+      _ <- peer1.startRegularSync()
+      _ <- peer2.startRegularSync()
+      _ <- peer2.connectToPeers(Set(peer1.node))
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumber)
+      _ <- peer2.mineNewBlocks(100.milliseconds, 2)(IdentityUpdate)
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumber + 2)
+      _ <- peer2.addCheckpointedBlock(peer2.bl.getBestBlock().get)
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumber + 3)
+      _ <- peer1.addCheckpointedBlock(peer1.bl.getBestBlock().get)
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumber + 4)
+      _ <- peer1.mineNewBlocks(100.milliseconds, 1)(IdentityUpdate)
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumber + 5)
+    } yield {
+      assert(peer1.bl.getBestBlock().get.hash == peer2.bl.getBestBlock().get.hash)
+      assert(peer1.bl.getLatestCheckpointBlockNumber() == peer2.bl.getLatestCheckpointBlockNumber())
+    }
+  }
+
+  "peers should chose the branch with a checkpoint discarding blocks that come after the checkpoint" in customTestCaseResourceM(
+    FakePeer.start2FakePeersRes()
```

Comment:
I'm missing something understanding this test. I was expecting some mined blocks after the checkpointing creation

Reply:
the test describes a scenario when a node with 30 blocks, receives a signal of generating a checkpoint at height 25. Since the chain can't be amended, it means that 25 blocks are kept as is, 5 blocks are discarded, and 1 checkpoint at height 26 is generated